src/eureka/S3_data_reduction/bright2flux.py

```diff
@@ -84,7 +84,14 @@ def dn2electrons(data, meta):
     ny = data.attrs['mhdr']['SUBSIZE2']
 
     # Load gain array in units of e-/ADU
-    gain = fits.getdata(meta.gainfile)[ystart:ystart+ny, xstart:xstart+nx]
+    gain_header = fits.getheader(meta.gainfile)
+    xstart_gain = gain_header['SUBSTRT1']
+    ystart_gain = gain_header['SUBSTRT2']
+
+    ystart_trim = ystart-ystart_gain + 1  # 1 indexed, NOT zero                                                         
+    xstart_trim = xstart-xstart_gain + 1
+
+    gain = fits.getdata(meta.gainfile)[ystart_trim:ystart_trim+ny, xstart_trim:xstart_trim+nx]
```

Comment:
Oh my, this PR just made me realize that a few lines below we rotate the MIRI gain file but don't then do `[:, ::-1]` like we do with the data, so our gain file is definitely being applied wrong for MIRI!